nest/proto.c

```diff
@@ -89,12 +97,58 @@ proto_log_state_change(struct proto *p)
 	{
 	  p->last_state_name_announced = name;
 	  PD(p, "State changed to %s", proto_state_name(p));
+	  notify_bgp_stat_daemon(p);
 	}
     }
   else
     p->last_state_name_announced = NULL;
 }
 
+/**
+ * @proto: pointer to struct proto
+ * Function sends a UDP message to the BGP stat daemon notifying the state of the router
+ * This function will be called whenever the router state changes
+ */
+static void
+notify_bgp_stat_daemon(struct proto *p){
+  if (!p)
+  {
+    return;
+  }
+  char * state = proto_state_name(p);
+  char * name = p->name;
+  char * message = (char*)malloc(strlen(state) + strlen(name)+ 20);
```

Comment:
I'm not sure what sandy would think is best, but we might want the packet to be structured a little big more

we might want to have it start with a version number, message type followed by state length and followed by name length.

Even if we don't want to do a more traditional type of udp packet, then we should still ensure a message type is encoded some where